### PR TITLE
Handling merge commits

### PR DESCRIPTION
**Pull request by parkouss**
See https://parkouss.wordpress.com/2015/11/14/mozregression-new-way-for-handling-merges/

### CODE REVIEW

File: gui/mozregui/bisection.py [hunk 1]
```diff
@@ -360,24 +372,38 @@ def bisection_finished(self, bisection, resultcode):
...
         dialog(self.mainwindow, "End of the bisection", msg)
         self.stop()
+
+    @Slot(object, str, str, str)
+    def handle_merge(self, bisection, branch, good_rev, bad_rev):
+        if QMessageBox.question(
+                self.mainwindow,
+                "Merge found",
+                "Found a merge from %s. Do you want to bisect further"
+                " in there ?" % branch,
```
Comment (wlach) [inline gui/mozregui/bisection.py]: I'd remove the `in there`

File: mozregression/bisector.py [hunk 2]
```diff
@@ -276,6 +228,66 @@ def user_exit(self, mid):
...
 
+    def handle_merge(self):
+        # let's check if we are facing a merge, and in that case,
+        # continue the bisection from the merged branch.
+        result = None
+
+        self._logger.debug("Starting merge handling...")
+        # so, we have to check the commit of the interesting push
+        # that is the most recent push
+        interesting_push = self.build_range[1]
```
Comment (wlach) [inline mozregression/bisector.py]: I really don't get what "interesting" means here. It just seems to be the end of the build range? Maybe call it `most_recent_push`?

File: mozregression/json_pushes.py [hunk 3]
```diff
@@ -45,18 +41,18 @@ def _request(self, url, check_changeset=None):
...
-    def pushlog_for_change(self, changeset):
+    def pushlog_for_change(self, changeset, **kwargs):
         """
         Returns the json pushlog object that match the given changeset.
 
         A MozRegressionError is thrown if None is found.
         """
         return next(self._request(
-            self.json_pushes_url(changeset=changeset),
+            self.json_pushes_url(changeset=changeset, **kwargs),
```
Comment (wlach) [inline mozregression/json_pushes.py]: You might be able to save the call to json_pushes_url by using request's builtin support for taking parameters. We do this in the treeherder client: 

https://github.com/mozilla/treeherder/blob/master/treeherder/client/thclient/client.py#L686

File: mozregression/main.py [hunk 4]
```diff
@@ -155,6 +149,13 @@ def _bisect_inbounds(self, good_rev, bad_rev):
                     "It seems that you used two changesets that are in"
                     " in the same push. Check the pushlog url."
                 )
+            elif len(handler.build_range) == 2:
+                # range reduced to 2 push, that is, one good, one bad.
```
Comment (wlach) [inline mozregression/main.py]: 2 push -> 2 pushes.
that is, -> that is:

File: mozregression/bisector.py (hunk 2 above)
Comment (parkouss) [inline mozregression/bisector.py]: totally, thanks!

**Pull request merged**